diff -rupN ltib.fsl/bin/Ltibutils.pm ltib/bin/Ltibutils.pm
--- ltib.fsl/bin/Ltibutils.pm	2010-05-26 04:45:24.000000000 +0300
+++ ltib/bin/Ltibutils.pm	2012-03-14 16:14:26.893239670 +0200
@@ -560,7 +560,7 @@ $app_checks = {
     binutils         => 'ar --version 2>/dev/null',
     'gcc-c++'        => 'g++ --version 2>/dev/null',
     glibc            => 'ldd --version 2>/dev/null',
-    'glibc-devel'    => sub { -f '/usr/lib/libm.so' || -f '/usr/lib64/libz.so'},
+    'glibc-devel'    => sub { -f '/usr/lib/libm.so' || -f '/usr/lib64/libz.so' || -f '/usr/lib/i386-linux-gnu/libm.so'},
     'glibc-headers'  => sub { -f '/usr/include/stdio.h' },
     'libstdc++' => sub {
             return system_nb(<<TXT) == 0;
@@ -582,7 +582,8 @@ TXT
                     },
     zlib         => sub { my @f = (glob('/usr/lib/libz.so*'),
                                    glob('/lib/libz.so*'),
-                                   glob('/lib64/libz.so*')  ); @f > 1 ? 1 : 0 },
+                                   glob('/lib64/libz.so*'),
+                                   glob('/lib/i386-linux-gnu/libz.so*')  ); @f > 1 ? 1 : 0 },
     'zlib-devel' => sub { -f '/usr/include/zlib.h' },
 };
 
diff -rupN ltib.fsl/ltib ltib/ltib
--- ltib.fsl/ltib	2010-05-26 04:45:24.000000000 +0300
+++ ltib/ltib	2012-03-14 16:48:56.251501091 +0200
@@ -926,6 +926,7 @@ TXT
             $cmd  = "$cf->{sudo} $cf->{rpm} ";
             $cmd .= "--root $cf->{rpmroot} ";
             $cmd .= "--dbpath $cf->{rpmdb} ";
+            $cmd .= "--force-debian " if $rpm =~ m/rpm-fs/ && `uname -a` =~ m/ubuntu/i;
             $cmd .= "--prefix $cf->{rpmipfx} " if $cf->{rpmipfx};
             $cmd .= "--ignorearch -ivh ";
             $cmd .= "--force "  unless $cf->{conflicts} || $cf->{hostinst};

