--- ltib.orig/bin/Ltibutils.pm	2012-04-30 01:48:21.000000000 +0200
+++ ltib/bin/Ltibutils.pm	2012-04-30 01:48:59.000000000 +0200
@@ -555,12 +555,15 @@
     return 0;
 }
 
-
 $app_checks = {
     binutils         => 'ar --version 2>/dev/null',
     'gcc-c++'        => 'g++ --version 2>/dev/null',
     glibc            => 'ldd --version 2>/dev/null',
-    'glibc-devel'    => sub { -f '/usr/lib/libm.so' || -f '/usr/lib64/libz.so'},
+    'glibc-devel'    => sub { -f '/usr/lib/libm.so' 
+                           || -f '/usr/lib64/libz.so'
+                           || -f '/usr/lib/i386-linux-gnu/libm.so'
+                           || -f '/usr/lib/x86_64-linux-gnu/libm.so'
+                       },
     'glibc-headers'  => sub { -f '/usr/include/stdio.h' },
     'libstdc++' => sub {
             return system_nb(<<TXT) == 0;
@@ -581,11 +584,17 @@
                         `makeinfo --version 2>/dev/null`;
                     },
     zlib         => sub { my @f = (glob('/usr/lib/libz.so*'),
+                                   glob('/usr/lib64/libz.so*'),
                                    glob('/lib/libz.so*'),
-                                   glob('/lib64/libz.so*')  ); @f > 1 ? 1 : 0 },
+                                   glob('/lib64/libz.so*'),
+                                   glob('/lib/i386-linux-gnu/libz.so*'),
+                                   glob('/usr/lib/x86_64-linux-gnu/libz.so*'),
+                                   glob('/lib/x86_64-linux-gnu/libz.so*')); 
+                                   @f > 1 ? 1 : 0 },
     'zlib-devel' => sub { -f '/usr/include/zlib.h' },
 };
 
+
 sub get_ver
 {
     my ($pkg) = @_;
