diff --git a/dist/lfs-5.1/wget/wget.spec b/dist/lfs-5.1/wget/wget.spec
index 27ed934..d84b2bd 100755
--- a/dist/lfs-5.1/wget/wget.spec
+++ b/dist/lfs-5.1/wget/wget.spec
@@ -19,7 +19,7 @@ Prefix          : %{pfx}
 %setup 
 
 %Build
-./configure --prefix=%{_prefix} --sysconfdir=%{_sysconfdir} --host=$CFGHOST --mandir=%{_mandir}
+./configure --prefix=%{_prefix} --sysconfdir=%{_sysconfdir} --host=$CFGHOST --mandir=%{_mandir} --without-ssl
 make
 
 %Install
