diff --git a/dist/lfs-5.1/sparse/sparse.spec b/dist/lfs-5.1/sparse/sparse.spec
index 5babbb2..15c1c86 100755
--- a/dist/lfs-5.1/sparse/sparse.spec
+++ b/dist/lfs-5.1/sparse/sparse.spec
@@ -9,6 +9,7 @@ License		: Open Software License
 URL		: http://kernel.org/pub/software/devel/sparse/
 Source0		: %{name}-%{version}.tar.gz
 Patch0          : %{name}-%{version}-array-fix.patch
+Patch1          : sparse-0.4-1316623824.patch
 BuildRoot       : %{_tmppath}/%{name}
 Prefix          : %{pfx}
 
@@ -31,6 +32,7 @@ Sparse is primarily used in the development and debugging of the Linux kernel.
 %prep
 %setup -q
 %patch0 -p1
+%patch1 -p1
 
 %build
 make DESTDIR=$RPM_BUILD_ROOT PREFIX="%{pfx}/%{_prefix}" CFLAGS="%{optflags}"
