diff --exclude CVS --exclude .git -uNr sparse-0.4/parse.h sparse-0.4.modified/parse.h
--- sparse-0.4/parse.h	2007-09-15 15:56:10.000000000 -0400
+++ sparse-0.4.modified/parse.h	2011-09-21 12:49:35.586591006 -0400
@@ -35,10 +35,6 @@
 		struct /* declaration */ {
 			struct symbol_list *declaration;
 		};
-		struct /* label_arg */ {
-			struct symbol *label;
-			struct statement *label_statement;
-		};
 		struct {
 			struct expression *expression;
 			struct expression *context;
Binary files sparse-0.4/target.o and sparse-0.4.modified/target.o differ
Binary files sparse-0.4/test-lexing.o and sparse-0.4.modified/test-lexing.o differ