--- plugins/input-raw.c 2013-06-23 19:54:23.000000000 +0000
+++ plugins/input-raw.c	2013-06-23 19:54:43.000000000 +0000
@@ -43,6 +43,7 @@
 #ifndef ABS_MT_POSITION_X
 # define ABS_MT_POSITION_X       0x35    /* Center X ellipse position */
 # define ABS_MT_POSITION_Y       0x36    /* Center Y ellipse position */
+# define ABS_MT_POSITION_MAJOR	 0x30    /* Press */
 #endif
 
 #include "tslib-private.h"
@@ -98,8 +99,8 @@
 	if ((ioctl(ts->fd, EVIOCGBIT(EV_ABS, sizeof(absbit)), absbit)) < 0 ||
 		!(absbit[BIT_WORD(ABS_X)] & BIT_MASK(ABS_X)) ||
 		!(absbit[BIT_WORD(ABS_Y)] & BIT_MASK(ABS_Y))) {
-		fprintf(stderr, "tslib: Selected device is not a touchscreen (must support ABS_X and ABS_Y events)\n");
-		return -1;
+		/*fprintf(stderr, "tslib: Selected device is not a touchscreen (must support ABS_X and ABS_Y events)\n");
+		return -1;*/
 	}
 
 	/* Since some touchscreens (eg. infrared) physically can't measure pressure,
@@ -203,6 +204,9 @@
 				case ABS_MT_POSITION_Y:
 					i->current_y = ev.value;
 					break;
+                case ABS_MT_TOUCH_MAJOR:
+                    i->current_p = ev.value;
+                    break;
 				case ABS_PRESSURE:
 					i->current_p = ev.value;
 					break;
